Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM iojs:latest
+
+
+#
+# installation
+#
+
+RUN mkdir /app
+RUN npm install -g grunt-cli coffee-script http-server
+
+ADD ./package.json /app/package.json
+RUN cd /app && npm install -q
```

Comment:
You're copying everything to the `/app` folder some lines below, so I think you can avoid explicitly adding the `package.json` there and simply run `npm install -q` after you cd to the `/app` folder (i.e.`RUN cd /app && npm install -q && grunt build`).